build/BUILD.bazel

```diff
@@ -75,6 +88,10 @@ grep ^STABLE_BUILD_SCM_REVISION bazel-out/stable-status.txt \
     stamp = 1,
 )
 
+# Building this rule will result in invocation of multi_arch_container rule because
+# the srcs list contains Bazel labels that correspond to the output of the
+# native.genrule in multi_arch_container definiton.
+# This rule will produce tarballs and docker tags for all server images.
```

Comment:
Thanks for digging in to this and explaining how it works.
Maybe move this last sentence to the top, because it's the bit that makes sense to me.
The other comments I don't currently understand, but I get the feeling they will be useful if I ever have to dig into this process myself.